pandas/core/groupby/generic.py

```diff
@@ -1495,7 +1507,9 @@ def _transform_fast(self, result: DataFrame, func_nm: str) -> DataFrame:
                 res = maybe_cast_result(res, obj.iloc[:, i], how=func_nm)
             output.append(res)
 
-        return DataFrame._from_arrays(output, columns=result.columns, index=obj.index)
+        return type(self.obj)._from_arrays(
```

Comment:
why is this not self.obj._constructor like elsewhere?

Reply:
Good catch -- I don't think there's a reason for this to be any different. I made the change to use `self.obj._constructor._from_arrays()` for consistency/clarity and it passes all the `groupby` tests.